app/assets/javascripts/patients.coffee

```diff
@@ -22,13 +22,41 @@ calculateRemainder = ->
 valueToNumber = (val) ->
   +val || 0
 
+markFulfilledWhenFieldsChecked = ->
+  pledge_fields = [
+    '#patient_fulfillment_procedure_cost'
+    '#patient_fulfillment_check_number'
+    '#patient_fulfillment_gestation_at_procedure'
+    '#patient_fulfillment_date_of_check'
+    '#patient_fulfillment_procedure_date'
+  ]
+
+  i = 0
+  empty = true
+  el = $('#patient_fulfillment_fulfilled')
+
+  while i < pledge_fields.length
+    if $(pledge_fields[i]).val().length > 0
+      empty = false
+      if el.prop 'checked'
+        break;
+      else
+        el.prop 'checked', true
+      i++
+    else
+      i++
+  if empty == true
+        el.prop 'checked', false
+
+
 ready = ->
   $(document).on "click", "#toggle-call-log", ->
     $(".old-calls").toggleClass("hidden")
     html = if $(".old-calls").hasClass("hidden") then "View all calls" else "Limit list"
     $("#toggle-call-log").html(html)
 
   $(document).on "change", ".edit_patient", ->
+    markFulfilledWhenFieldsChecked()
```

Comment:
For patients where there isn't a pledge yet, this seems to be breaking the ability to save. To replicate: 

* run `rake db:seed`
* go to a patient who hasn't been marked pledge sent
* try and save '2017-10-10' in the appointment date input
* check the console and see the error on no `length` field

Maybe a good way around this is to pop into `_fulfillment`, give this form a specific id, and then only fire this action when that id changes?